src/lib/discord/commands/admin.ts

```diff
@@ -0,0 +1,215 @@
+import { prisma } from '$lib/db'
+import {
+  postDiscussion,
+  postAnswer,
+  markAnswered,
+  lockDiscussion,
+} from '$lib/github/queries'
+import { repositoriesWithDiscussions as repositories } from './_repositories'
+import { isThreadWithinHelpChannel } from '../support'
+import type {
+  GuildMember,
+  Message,
+  ChatInputCommandInteraction,
+  InteractionReplyOptions,
+  ThreadChannel,
+} from 'discord.js'
+
+const userIdToUsername = new Map<string, user>()
+
+// const iconMap = new Map<string, string>([
+//   ['Admin', "https://github.com/esauerbo1/Images/blob/main/Admin.svg"],
+//   ['Bot', 'https://github.com/esauerbo1/Images/blob/main/Bot.svg'],
+//   ['Amplify Bot Dev', 'https://github.com/esauerbo1/Images/blob/main/Amplify-Bot-Dev.svg'],
+//   ['Moderator', 'https://github.com/esauerbo1/Images/blob/main/Moderator.svg'],
+//   ['Amplify Staff', 'https://github.com/esauerbo1/Images/blob/main/Amplify-Staff.svg'],
+//   ['AWS Staff', 'https://github.com/esauerbo1/Images/blob/main/AWS-Staff.svg'],
+//   ['Community Builder', 'https://github.com/esauerbo1/Images/blob/main/Community-Builder.svg'],
+//   ['Contributor', 'https://github.com/esauerbo1/Images/blob/main/Contributor.svg'],
+//   ['Meetup Organizer', 'https://github.com/esauerbo1/Images/blob/main/Meetup-Organizer.svg'],
+//   ['Amplify Guru', 'https://github.com/esauerbo1/Images/blob/main/Amplify-Guru.svg'],
+//   ['@everyone', 'https://github.com/esauerbo1/Images/blob/main/%40everyone.svg'],
+// ])
+
+type user = {
+  username: string
+  avatar: string
+  highestRole: string
+}
+
+function getUser(userId: string, guildMember: GuildMember | null) {
+  if (!userIdToUsername.get(userId)) {
+    let role = '@everyone'
+    if (guildMember?.roles?.highest?.name) role = guildMember.roles.highest.name
+    userIdToUsername.set(`${userId}`, {
+      username: `${faker.unique(faker.color.human)} ${faker.unique(
+        faker.hacker.noun
+      )}`,
+      avatar: `https://avatars.dicebear.com/api/bottts/${userId}.svg`,
+      highestRole: `(${role})`,
+    })
+  }
+
+  const user = userIdToUsername.get(userId)
+  return `<img src=${user?.avatar} width="30" style="vertical-align:bottom;display:inline" /> **${user?.username}** ${user?.highestRole}: `
+}
+
+function createAnswer(answer: Message) {
+  const user = getUser(answer.author.id, answer.member)
+  return `${user} ${answer.content}`
+}
+
+function createDiscussionBody(
+  firstMessage: Message,
+  messages: Map<string, Message>,
+  threadUrl: string
+): string {
+  let user = getUser(firstMessage.author.id, firstMessage.member)
+  let body = `${user} ${firstMessage.content}\n\n`
+  Array.from(messages.values())
+    .reverse()
+    .filter((message: Message<boolean>) => !message.author.bot)
+    .forEach(async (message: Message<boolean>) => {
+      user = getUser(message.author.id, message.member)
+      body += `${user} ${message.content}\n\n`
+    })
+  body += `#### 🕹️ View the original Discord thread [here](${threadUrl})\n`
```

Comment:
minor suggestion, how do you feel about separating (with `---`) the link content from the conversation content?

```suggestion
  body += `---\n🕹️ [View the original Discord thread](${threadUrl})\n`
```